src/placement/pl_map.c

```diff
@@ -558,11 +558,9 @@ pl_select_leader(daos_obj_id_t oid, uint32_t shard_idx, int shards_nr,
 	int				 start;
 	int				 pos;
 	int				 off;
+	int				 replica_idx;
 	int				 i;
 
-	if (shards_nr <= shard_idx)
-		return -DER_INVAL;
-
```

Comment:
Please rename the @shards_nr as @grp_size, and update the function comment.

Reply:
ok, I refreshed before see it, let me refresh it again.
now it can pass the test (of DAOS-2743) without your PR694.